### ISSUE TITLE
Incorrectly skipped XDS push when Istio CRDs are changed

### ISSUE THREAD

**Issue by weibohe** [istio/istio#29414]
(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**
When Istio CRDs [in this list](https://github.com/airbnb/istio/blob/master/pilot/pkg/model/sidecar.go#L42-L47) is defined in workload namespace and Sidecar CRD is defined in root namespace, `DependsOnConfig` will incorrectly return false when CRDs are changed, causing XDS Push to be skipped.  
```
	// This kind of config will trigger a change if made in the root namespace or the same namespace
	if _, f := sidecarScopeNamespaceConfigTypes[config.Kind]; f {
		return config.Namespace == sc.RootNamespace || config.Namespace == sc.Config.Namespace
	}
```
config.Namespace will be the workload namespace in this case. 
Sidecar namespace will be the root namespace (istio-system). 

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
XDS push will be triggered correctly. 

**Steps to reproduce the bug**

1. Define a default Sidecar CRD in istio-system which 
```
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
metadata:
  name: default
  namespace: istio-system
spec:
  egress:
  - hosts:
    - ./*
```
2. Create a workload namespace with any workload and define an Authorization Policy inside that namespace.
3. Change the authz policy.
4. Monitor in Istiod logs that the update is skipped.
`2020-12-04T18:10:47.652820Z	debug	ads	Skipping push to sidecar~100.64.11.76~fortioclient-6bb9b784c6-f2ktw.airmesh-test-authz~airmesh-test-authz.svc.cluster.local-184, no updates required`
5. After defining a sidecar CRD in the same workload namespace, XDS pushes are successfully triggered (But this is a workaround as some namespace would like to inherit the default Sidecar CRD from root namespace). 

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
1.7.5 & master

**How was Istio installed?**
istioctl

**Environment where the bug was observed (cloud vendor, OS, etc)**
Kubernetes 

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by hzxuzhonghu** [comment]
Thanks for reporting.

**Comment by weibohe** [comment]
@hzxuzhonghu thanks for the quick fix! Any chance we can backport the fix to 1.7?
